tests/countries/test_nepal.py

```diff
@@ -0,0 +1,522 @@
+        dt = (
+            "2019-08-24",
+            "2020-08-12",
+        )
+        self.assertHolidayName(name, self.workday_holidays, dt)
+        self.assertNoHolidayName(name, dt)
+
+        name = "Lakshmi Puja"
+        dt = (
+            "2019-10-27",
+            "2020-11-14",
+            "2021-11-04",
+            "2022-10-24",
+            "2023-11-12",
+            "2024-11-01",
+            "2025-10-20",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2036))
+
+        name = "Maha Ashtami"
+        dt = (
+            "2019-10-06",
+            "2020-10-23",
+            "2021-10-13",
+            "2022-10-03",
+            "2023-10-22",
+            "2024-10-10",
+            "2025-09-30",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2036))
+
+        name = "Maha Navami"
+        dt = (
+            "2019-10-07",
+            "2020-10-24",
+            "2021-10-14",
+            "2022-10-04",
+            "2023-10-23",
+            "2024-10-11",
+            "2025-10-01",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2036))
+
+        name = "Maghe Sankranti"
+        dt = (
+            "2019-01-15",
+            "2020-01-15",
+            "2021-01-14",
+            "2022-01-14",
+            "2023-01-14",
+            "2024-01-14",
+            "2025-01-14",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2036))
+
+        name = "Mha Puja"
+        dt = (
+            "2019-10-28",
+            "2020-11-15",
+            "2021-11-05",
+            "2022-10-25",
+            "2023-11-13",
+            "2024-11-02",
+            "2025-10-22",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2036))
+
+        name = "Nepali New Year (Vikram Sambat)"
+        dt = (
+            "2019-04-14",
+            "2020-04-13",
+            "2021-04-14",
+            "2022-04-14",
+            "2023-04-14",
+            "2024-04-13",
+            "2025-04-13",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2036))
+
+        name = "Phulpati"
+        dt = (
+            "2019-10-05",
+            "2020-10-22",
+            "2021-10-12",
+            "2022-10-02",
+            "2023-10-21",
+            "2024-10-09",
+            "2025-09-29",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2036))
+
+        name = "Ram Navami"
+        dt = (
+            "2021-04-21",
+            "2022-04-10",
+            "2023-03-30",
+            "2024-04-17",
+            "2025-04-06",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2001, 2019))
+        self.assertHolidayName(name, range(2021, 2036))
```

Comment:
We can combine this together, same goes for elsewhere 
```suggestion
        self.assertHolidayName(name, range(2001, 2019), range(2021, 2036))
```

I think we should also break down the test cases into individual ones instead of having a single large `test_ranged_holidays`. Maybe something like `test_ram_navami`, `test_phulpati`, `test_nepali_new_year`, etc. would be more manageable.

Reply:
Yes, I just updated the code to handle multiple ranges.  
I'll update the rest of the functions into individual test cases as you suggested asap.